scheduler/src/cook/kubernetes/api.clj

```diff
@@ -686,20 +696,25 @@
     :or {pod-priority-class cook-job-pod-priority-class
          pod-supports-cook-init? true
          pod-supports-cook-sidecar? true}}]
-  (let [{:keys [scalar-requests job]} task-request
+  (let [{:keys [scalar-requests job resources]} task-request
         ;; NOTE: The scheduler's adjust-job-resources-for-pool-fn may modify :resources,
         ;; whereas :scalar-requests always contains the unmodified job resource values.
         {:strs [mem cpus]} scalar-requests
         {:keys [docker volumes]} container
         {:keys [image parameters port-mapping]} docker
         {:keys [environment]} command
+        pool-name (some-> job :job/pool :pool/name)
+        ; gpu count is not stored in scalar-requests
+        gpus (or (:gpus resources) 0)
+        gpu-model-requested (when (pos? gpus)
+                              (or (-> job :job/environment get-gpu-model-from-task-metadata)
+                                  (util/match-based-on-pool-name (config/valid-gpu-models) pool-name :default-model)))
```

Comment:
We have these same 3 lines in the `gpu-host-constraint`. Should we extract a function and use it both places?

Reply:
sure, how does `get-gpu-model-requested` sound?

Reply:
How about `job->gpu-model-requested`?

Reply:
sounds good to me